CMakeLists.txt

```diff
@@ -203,35 +203,69 @@ get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
 if(TRITON_BUILD_PYTHON_MODULE)
   add_library(triton SHARED ${PYTHON_SRC})
 
-  target_link_libraries(triton
-    TritonAnalysis
-    TritonTransforms
-    TritonGPUTransforms
-    TritonLLVMIR
-    TritonPTX
-    ${dialect_libs}
-    ${conversion_libs}
-    # optimizations
-    MLIRPass
-    MLIRTransforms
-    MLIRLLVMDialect
-    MLIRSupport
-    MLIRTargetLLVMIRExport
-    MLIRExecutionEngine
-    MLIRMathToLLVM
-    MLIRNVVMToLLVMIRTranslation
-    MLIRIR
-  )
-
-  target_link_options(triton PRIVATE ${LLVM_LDFLAGS})
-
   if(WIN32)
-      target_link_libraries(triton PRIVATE ${LLVM_LIBRARIES} dl) # dl is from dlfcn-win32
+    target_link_libraries(triton PRIVATE ${LLVM_LIBRARIES} ${CMAKE_DL_LIBS}
+      TritonAnalysis
+      TritonTransforms
+      TritonGPUTransforms
+      TritonLLVMIR
+      TritonPTX
+      ${dialect_libs}
+      ${conversion_libs}
+      # optimizations
+      MLIRPass
+      MLIRTransforms
+      MLIRLLVMDialect
+      MLIRSupport
+      MLIRTargetLLVMIRExport
+      MLIRExecutionEngine
+      MLIRMathToLLVM
+      MLIRNVVMToLLVMIRTranslation
+      MLIRIR
+    )
   elseif(APPLE)
-      target_link_libraries(triton ${LLVM_LIBRARIES} z)
+    target_link_libraries(triton ${LLVM_LIBRARIES} z
+      TritonAnalysis
+      TritonTransforms
+      TritonGPUTransforms
+      TritonLLVMIR
+      TritonPTX
+      ${dialect_libs}
+      ${conversion_libs}
+      # optimizations
+      MLIRPass
+      MLIRTransforms
+      MLIRLLVMDialect
+      MLIRSupport
+      MLIRTargetLLVMIRExport
+      MLIRExecutionEngine
+      MLIRMathToLLVM
+      MLIRNVVMToLLVMIRTranslation
+      MLIRIR
+    )
   else()
-      target_link_libraries(triton ${LLVM_LIBRARIES} z stdc++fs)
+    target_link_libraries(triton ${LLVM_LIBRARIES} z stdc++fs
+      TritonAnalysis
+      TritonTransforms
+      TritonGPUTransforms
+      TritonLLVMIR
+      TritonPTX
+      ${dialect_libs}
+      ${conversion_libs}
+      # optimizations
+      MLIRPass
+      MLIRTransforms
+      MLIRLLVMDialect
+      MLIRSupport
+      MLIRTargetLLVMIRExport
+      MLIRExecutionEngine
+      MLIRMathToLLVM
+      MLIRNVVMToLLVMIRTranslation
+      MLIRIR
```

Comment:
that's a lot of code duplication. Could you use a variable for libs that are reused?